libcasr/src/report.rs

```diff
@@ -774,3 +774,230 @@ pub fn cluster_reports(casreps: &[CrashReport]) -> Result<Vec<u32>> {
 
     cluster_stacktraces(&traces)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::constants::*;
+    use crate::init_ignored_frames;
+
+    #[test]
+    fn test_report_display() {
+        // TODO: Fill all empty fields
+        let report = CrashReport {
+            proc_fd: vec![],
+            network_connections: vec![],
+            registers: Registers::new(),
+            package: "".to_string(),
+            package_version: "".to_string(),
+            package_architecture: "".to_string(),
+            package_description: "".to_string(),
+            crashline: "/gcc-build/x86_64-pc-linux-gnu/libstdc++-v3/include/bits/basic_string.h:187:28".to_string(),
+            date: "2023-09-29T15:31:39.818262+03:00".to_string(),
+            uname: "Linux astra-stand 5.15.0-84-generic #93-Ubuntu SMP Tue Sep 5 17:16:10 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux".to_string(),
+            os: "Ubuntu".to_string(),
+            os_release: "22.04".to_string(),
+            architecture: "amd64".to_string(),
+            executable_path: "/home/hkctkuy/github/casr/casr/tests/casr_tests/bin/load_fuzzer".to_string(),
+            proc_environ: vec!["CARGO=/home/hkctkuy/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo".to_string()],
+            proc_cmdline: "/home/hkctkuy/github/casr/casr/tests/casr_tests/bin/load_fuzzer /home/hkctkuy/github/casr/casr/tests/casr_tests/casrep/libfuzzer_crashes_xlnt/crash-49c4e6270849ab5c9c13f64b98f817efa840d8a3".to_string(),
+            stdin: "/home/hkctkuy/github/casr/casr/tests/casr_tests/casrep/afl-out-xlnt-small/afl_main-worker/crashes/id:000007,sig:00,sync:afl_s01-worker,src:000371".to_string(),
+            proc_status: vec!["process 2945397".to_string()],
+            proc_maps: vec![
+                "      0x555555554000     0x555555556000     0x2000        0x0 /usr/local/bin/tiff2pdf".to_string(),
+            ],
+            stacktrace: vec![
```

Comment:
You could decrease the size of vec![] of up to two elements, for all `*_report` fields too.

Reply:
Done